types/hubot/index.d.ts

```diff
@@ -40,6 +41,8 @@ declare namespace Hubot {
         readonly adapter: A;
 
         constructor(adapterPath: string, adapter: string, httpd: boolean, name: string, alias?: string);
+        catchAll(callback: ListenerCallback<this>): void;
+        catchAll(options: any, callback: ListenerCallback<this>): void;
```

Comment:
Can options be optional here instead?

Reply:
Yes, and it can be optional for the hear and response methods as well, should I change them all? :)

Reply:
It would be really cool if you did :)

Reply:
I realize now that I might have been wrong. I'm quite new to TypeScript, the options parameter is always optional in the JS code, but for TypeScript, optional parameters needs to come after mandatory parameters. I don't know how to make the options parameter optional, if possible...

Reply:
Putting `options?:` would make it optional, though optional parameters do need to come after all non-optional parameters. [Optional parameters vs overloads here](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-optional-parameters). There may have been some reason you did not want to do this here? Probably not a big deal either way.